src/core/thread/discover_scanner.hpp

```diff
@@ -163,6 +180,10 @@ class DiscoverScanner : public InstanceLocator
     uint8_t          mScanChannel;
     bool             mEnableFiltering : 1;
     bool             mShouldRestorePanId : 1;
+    uint32_t         mOui;
+    const uint8_t *  mAdvData;
+    uint8_t          mAdvDataLength;
```

Comment:
```suggestion
    JoinerAdvertisementTlv mJoinerAdvertisement;
```
Use a `JoinerAdvertisementTlv` instead?

Reply:
this is intended to save memory.

Reply:
Fine with both approach. 
But actually only saved 1 byte (i.e. Type of TLV) ?